### PR TITLE
Install phpstan extension also for root package

### LINKED ISSUE #36: composer.json's `extra` section must be honored also for root-package
### Currently state

composer root package requires manual `include`.

### Expected state

all extentions exported using composer.json's `extra` section must be used for the current/root repo too

### PR DESCRIPTION
**Pull request by mvorisek**
fixes https://github.com/phpstan/extension-installer/issues/36

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "require": {
         "php": "^7.1 || ^8.0",
         "composer-plugin-api": "^1.1 || ^2.0",
-        "phpstan/phpstan": ">=0.11.6"
+        "phpstan/phpstan": ">=0.12.19"
```
Comment (mvorisek) [inline composer.json:11]: to support infile `@phpstan-ignore-line`

File: src/Plugin.php [hunk 2]
```diff
@@ -133,12 +135,10 @@ public function process(Event $event): void
 		file_put_contents($generatedConfigFilePath, $generatedConfigFileContents);
 		$io->write('<info>phpstan/extension-installer:</info> Extensions installed');
 
-		if ($oldGeneratedConfigFileHash === md5($generatedConfigFileContents)) {
-			return;
-		}
-
-		foreach (array_keys($installedPackages) as $name) {
-			$io->write(sprintf('> <info>%s:</info> installed', $name));
+		if ($io->isVerbose()) {
```
Comment (mvorisek) [inline src/Plugin.php:138]: previously, the logging was not repeatable (based on if file hash has changed or not), now it is repeatable and installed packages are dumped only if verbose logging is enabled

**Comment by mvorisek** [comment]
PR is done, `Build / PHPStan (8.0, highest)` CI error seems unrelated

**Comment by mvorisek** [comment]
@ondrejmirtes can you please review this PR?

**Comment by mvorisek** [comment]
> 1. This PR does too many unrelated changes

I have separated https://github.com/phpstan/extension-installer/pull/42 , all other changes are related

>     2. I'm not even sure that [composer.json's `extra` section must be honored also for root-package #36](https://github.com/phpstan/extension-installer/issues/36) is a real problem that needs solving. The whole point of this package is that it saves a few lines in your phpstan.neon.

Exactly that is the point. When phpstan config is added to composer.json, it must always apply.

**Comment by mvorisek** [comment]
@ondrejmirtes can you please review?

**Comment by ondrejmirtes** [comment]
I'm still not sure about the implications and the usefulness. I'd rather not do it.

**Comment by mvorisek** [comment]
yes, repos including some extra conf manually need to be updated (otherwise phpstan will throw for config included twice), but this PR unifies how the composer config behaves - like for ex. `autoload` config - it works for root project and also when included

other than this change needed, do you have any other implication in head?

to maintain compatibility with existing projects, what about releasing this as 2.x version?

**Comment by ondrejmirtes** [comment]
I feel like there isn't any demand for this, and I feel against it too. So closing. Thanks for understanding.

**Pull request closed without merging**